checks.d/mysql.py

```diff
@@ -810,9 +820,28 @@ def _get_replica_stats(self, db):
         with closing(db.cursor()) as cursor:
             cursor.execute("SHOW SLAVE STATUS;")
             replica_results = dict(cursor.fetchall())
+            cursor.execute("SHOW MASTER STATUS;")
+            binlog_results = dict(cursor.fetchall())
+            if binlog_results:
+                replica_results.update({'Binlog_enabled': True})
 
             return replica_results
 
+    def _get_slave_status(self, db, nonblocking=False):
+        with closing(db.cursor()) as cursor:
+            # querying threads instead of PROCESSLIST to avoid mutex impact on
+            # performance.
+            if nonblocking:
+                cursor.execute("SELECT THREAD_ID, NAME FROM performance_schema.threads WHERE NAME LIKE '%worker'")
+            else:
+                cursor.execute("SELECT * FROM INFORMATION_SCHEMA.PROCESSLIST WHERE COMMAND LIKE '%Binlog dump%'")
```

Comment:
Where should these be run from? The master or a slave?

Just want to make sure I understand exactly how we're choosing the status we send for the service check, both on the master and on a slave.